cetmix_tower_yaml/wizards/cx_tower_yaml_import_wiz.py

```diff
@@ -276,22 +287,27 @@ def _extract_secret_names(self, data: dict) -> list:
         def _recursive_extract(node):
             """Recursively extract secret names from nested structures."""
             if isinstance(node, dict):
+                # Extract from secret_ids
                 if "secret_ids" in node and isinstance(node["secret_ids"], list):
                     for item in node["secret_ids"]:
-                        if isinstance(item, dict) and "name" in item:
-                            secret_names.add(item["name"])
+                        if isinstance(item, dict) and "key_id" in item:
```

Comment:
What's the reason behind this change? Please provide some YAML code examples that require such approach.